### ISSUE TITLE
Test-extended.system-JDK8-aix_ppc-64_cmprssptrs SharedClasses.SCM23.MultiThread_0 crash Compiled_method fastIdentityHashCode

### ISSUE THREAD

**Issue by pshipton** [eclipse/openj9#4334]
https://ci.eclipse.org/openj9/job/Test-extended.system-JDK8-aix_ppc-64_cmprssptrs/125
```
MT1 stderr Unhandled exception
MT1 stderr Type=Segmentation error vmState=0x00000000
MT1 stderr J9Generic_Signal_Number=00000004 Signal_Number=0000000b Error_Value=00000000 Signal_Code=00000032
MT1 stderr Handler1=09001000A2772300 Handler2=09001000A2748BF0
MT1 stderr R0=00000000FFF58C78 R1=00000001100452C0 R2=00000000300213F0 R3=00000000FFF00250
MT1 stderr R4=0000000030008200 R5=000000000000000E R6=00000000FFF00250 R7=09001000A27D0F30
MT1 stderr R8=0000000000000000 R9=0000000000000000 R10=0000000000000002 R11=090000004B255D04
MT1 stderr R12=0000000042008848 R13=000000011004E800 R14=00000000300C5590 R15=0000000030008200
MT1 stderr R16=0000000121112F18 R17=FFFFFFFFFFFFFFFF R18=0000000000000000 R19=0000000000000000
MT1 stderr R20=0000000110045510 R21=0000000000000000 R22=0000000000000000 R23=0000000000000000
MT1 stderr R24=0000000000000000 R25=09001000A2775DB0 R26=0000000000000000 R27=0000000000000000
MT1 stderr R28=0000000000000000 R29=00000000FFF06C58 R30=00000000E0011160 R31=0000000000000000
MT1 stderr IAR=0000000110529CF8 LR=090000004B2553B8 MSR=A00000000200D032 CTR=0000000110529CDC
MT1 stderr CR=2200888000000004 FPSCR=8202000000000000 XER=0000000482020000
MT1 stderr FPR0 0000000000000055 (f: 85.000000, d: 4.199558e-322)
MT1 stderr FPR1 c3e0000000000000 (f: 0.000000, d: -9.223372e+18)
MT1 stderr FPR2 bfe002294874d0e5 (f: 1215615232.000000, d: -5.002638e-01)
MT1 stderr FPR3 3fa05cc875900bba (f: 1972374400.000000, d: 3.195788e-02)
MT1 stderr FPR4 3fa0a06f209fb76a (f: 547338112.000000, d: 3.247401e-02)
MT1 stderr FPR5 3fda7aecbeba4e46 (f: 3199880704.000000, d: 4.137527e-01)
MT1 stderr FPR6 3fd5555555555611 (f: 1431655936.000000, d: 3.333333e-01)
MT1 stderr FPR7 3f514726f2cfbe13 (f: 4073700864.000000, d: 1.054562e-03)
MT1 stderr FPR8 bfb96bfae093ca0d (f: 3767781888.000000, d: -9.930389e-02)
MT1 stderr FPR9 40fb39456ab53250 (f: 1790259840.000000, d: 1.115083e+05)
MT1 stderr FPR10 407816625220ec6e (f: 1377889408.000000, d: 3.853990e+02)
MT1 stderr FPR11 3fbc73cd95d5986d (f: 2513803264.000000, d: 1.111420e-01)
MT1 stderr FPR12 3ff0000000000000 (f: 0.000000, d: 1.000000e+00)
MT1 stderr FPR13 4020800000000000 (f: 0.000000, d: 8.250000e+00)
MT1 stderr FPR14 0000000000000000 (f: 0.000000, d: 0.000000e+00)
MT1 stderr FPR15 0000000000000000 (f: 0.000000, d: 0.000000e+00)
MT1 stderr FPR16 0000000000000000 (f: 0.000000, d: 0.000000e+00)
MT1 stderr FPR17 0000000000000000 (f: 0.000000, d: 0.000000e+00)
MT1 stderr FPR18 0000000000000000 (f: 0.000000, d: 0.000000e+00)
MT1 stderr FPR19 0000000000000000 (f: 0.000000, d: 0.000000e+00)
MT1 stderr FPR20 0000000000000000 (f: 0.000000, d: 0.000000e+00)
MT1 stderr FPR21 0000000000000000 (f: 0.000000, d: 0.000000e+00)
MT1 stderr FPR22 0000000000000000 (f: 0.000000, d: 0.000000e+00)
MT1 stderr FPR23 0000000000000000 (f: 0.000000, d: 0.000000e+00)
MT1 stderr FPR24 0000000000000000 (f: 0.000000, d: 0.000000e+00)
MT1 stderr FPR25 0000000000000000 (f: 0.000000, d: 0.000000e+00)
MT1 stderr FPR26 0000000000000000 (f: 0.000000, d: 0.000000e+00)
MT1 stderr FPR27 0000000000000000 (f: 0.000000, d: 0.000000e+00)
MT1 stderr FPR28 0000000000000000 (f: 0.000000, d: 0.000000e+00)
MT1 stderr FPR29 0000000000000000 (f: 0.000000, d: 0.000000e+00)
MT1 stderr FPR30 0000000000000000 (f: 0.000000, d: 0.000000e+00)
MT1 stderr FPR31 0000000000000000 (f: 0.000000, d: 0.000000e+00)
MT1 stderr 
MT1 stderr Compiled_method=java/lang/J9VMInternals.fastIdentityHashCode(Ljava/lang/Object;)I
MT1 stderr Target=2_90_20190116_964 (AIX 7.1)
MT1 stderr CPU=ppc64 (8 logical CPUs) (0x200000000 RAM)
MT1 stderr ----------- Stack Backtrace -----------
MT1 stderr runCallInMethod+0x200 (0x090000004AD68F24 [libj9vm29.so+0x56f24])
MT1 stderr gpProtectedRunCallInMethod__FPv+0x48 (0x090000004AD77D2C [libj9vm29.so+0x65d2c])
MT1 stderr signalProtectAndRunGlue+0x28 (0x090000004AD71FAC [libj9vm29.so+0x5ffac])
MT1 stderr omrsig_protect+0x3cc (0x090000004B018E70 [libj9prt29.so+0x68e70])
MT1 stderr gpProtectAndRun+0xf8 (0x090000004AD720FC [libj9vm29.so+0x600fc])
MT1 stderr gpCheckCallin+0x100 (0x090000004AD77CA4 [libj9vm29.so+0x65ca4])
MT1 stderr callStaticVoidMethod+0x48 (0x090000004ADAE38C [libj9vm29.so+0x9c38c])
MT1 stderr J9_CreateJavaVM+0x278 (0x090000004AD3A6FC [libj9vm29.so+0x286fc])
MT1 stderr JNI_CreateJavaVM+0xa0c (0x090000004ACC0C90 [libjvm.so+0x28c90])
MT1 stderr JNI_CreateJavaVM+0x1c4 (0x090000004AC82848 [libjvm.so+0x24848])
MT1 stderr (0x0000000100003940 [java+0x3940])
MT1 stderr _pthread_body+0xf0 (0x0900000000526E14 [libpthreads.a+0x3e14])
```

**Comment by pshipton** [comment]
20x grinder https://ci.eclipse.org/openj9/view/Test/job/Test-Grinder/196/

**Comment by DanHeidinga** [comment]
FYI @gita-omr Can you have someone take a look at this?

**Comment by R-Santhir** [comment]
Is it possible to get access to the test source? Running this locally would speed up investigation @DanHeidinga @pshipton

**Comment by pshipton** [comment]
If only they were opened sourced ... oh wait.
The tests live here https://github.com/eclipse/openj9-systemtest
Clone the open j9 repo (i.e. `git clone git@github.com:eclipse/openj9.git`) and then follow the quick start instructions. You would run `SharedClasses.SCM23.MultiThread_0`.
https://github.com/eclipse/openj9/blob/master/test/README.md
Note you need a machine with the test prerequisite tools installed, also detailed from the previous link.

**Comment by pshipton** [comment]
I'm wondering if there is some non-obvious step that needs to be added since this is system step.
@smlambert

**Comment by pshipton** [comment]
Ah, systemtest has it's own quickstart guide https://github.com/eclipse/openj9-systemtest#unix

**Comment by R-Santhir** [comment]
Awesome thanks @pshipton

**Comment by smlambert** [comment]
I should add that if you do not want to get into the down'n'dirty details of all the different tests we run, I created this guide to rerunning the different types locally:  
https://github.com/eclipse/openj9/wiki/Reproducing-Test-Failures-Locally  which contains examples from each set, including a system test example:
https://github.com/eclipse/openj9/wiki/Reproducing-Test-Failures-Locally#run-sanity-system-tests-on-jdk10_x86-64_linux_openj9-sdk

**Comment by smlambert** [comment]
You can also run a test Grinder on the internal Jenkins instance to run this test, see Test_Grinder tab to find the job (instructions: https://github.com/AdoptOpenJDK/openjdk-tests/wiki/How-to-Run-a-Grinder-Build-on-Jenkins).  Cores from internal Grinder jobs get zipped up in the test artifacts and stored in artifactory (available for download via a link from the Grinder job).

**Comment by pshipton** [comment]
Since 40 grinders passed, I'm remove this from the 0.12 milestone.

**Comment by pshipton** [comment]
Another one https://ci.eclipse.org/openj9/job/Test-extended.system-JDK8-aix_ppc-64_cmprssptrs/148

**Comment by gita-omr** [comment]
@R-Santhir could you provide an update please?

**Comment by gita-omr** [comment]
Maybe since it's SharedClasses @dsouzai has some suggestions how to reproduce?

**Comment by R-Santhir** [comment]
I've managed to see a crash locally and get it to fail semi consistently with the following options:
`disableAsyncCompilation,count=1,optlevel=hot`

However in the original failure the compiled method was:
```Compiled_method=java/lang/J9VMInternals.fastIdentityHashCode(Ljava/lang/Object;)I ``` 

**but I am seeing:**
```Method_being_compiled=sun/util/locale/StringTokenIterator.nextDelimiter(I)I```

I have yet to reproduce the original error on my machine. I am trying to use different option sets locally, while trying to run a larger grinder set as well.

**Comment by dsouzai** [comment]
I don't have any suggestions in terms of reproducing the failure; in the past (eg the osx issue), I've had to recreate the test setup locally by more or less invoking the git/make commands done in the Jenkins job, and manually grinding it (because I'm not sure if the grinder job on Jenkins will save cores...). Once you can reproduce the crash, make a copy of the SCC file, since we can then use that file to consistently reproduce the failure.

Once you have bad SCC, try to narrow the bad method using `-Xaot:loadLimitFile=<path to file>`; this is basically a limit file but only for AOT loads. However, if this causes the timing to change, the problem may not show up.

You could also look at the core file (when you reproduce this) to see what's actually causing the crash, and then see where the bad value comes from. It might be some address we're trying to load from that was never relocated (eg https://github.com/eclipse/openj9/issues/4138#issuecomment-461046246). The way you can see if we missed a relocation is by running `-Xaot:aotrtDebugLevel=27,rtLog=rtLog -Xjit:compilationThreads=1`, this is is really only useful when you know which method has the missing relocation.

Once you've got to the point where you're able to reproduce this because you have a bad SCC on hand, feel free to get a hold of me and I can try with the investigation (since you might not be familiar with the output of the aot debug output),

@yanluo7 since you recently spent a lot of time trying to reproduce his issue, do you have any other suggestions in terms of how to improve the chances of locally reproducing a crash?

**Comment by dsouzai** [comment]
@R-Santhir why `optlevel=hot`? Was the failing method's opt level hot? If that's the case, then it wouldn't have been an AOT'd body.

**Comment by R-Santhir** [comment]
Only because I was able to get a failure with `optlevel=hot` and it wasn't even the original failing method. I am not sure what the optlevel for the original failure was.

 I'll try to reproduce a failure with the failing method from the original, so I can proceed with the SCC analysis you mentioned. Thanks @dsouzai

**Comment by gita-omr** [comment]
Should not KCA tell us opt level for the failing method?

**Comment by yanluo7** [comment]
For my issue, it was pretty intermittent to get a failure, so just need to keep at it; but once I had a failure make sure you save the SC somewhere.
Refer to https://github.com/eclipse/openj9/issues/4138#issuecomment-458292650 for some options I used.

**Comment by vij-singh** [comment]
Any new news on this one?

**Comment by gita-omr** [comment]
We have couple of ideas, hopefully something will come up. Basically, I suspect java/lang/J9VMInternals.fastIdentityHashCode(Ljava/lang/Object;)I is rarely compiled on its own, usually inlined. I hope if we enforce compilation of that method (eg disable inlining of it) we will see the failure. 

@jdmpapin, we meant to ask: do you know anything special about that method?

**Comment by pshipton** [comment]
https://ci.eclipse.org/openj9/job/Test-extended.system-JDK8-aix_ppc-64_cmprssptrs/160/

**Comment by pshipton** [comment]
The above failure is a little different `Compiled_method=java/lang/Object.hashCode()I

```
MT5 stderr Unhandled exception
MT5 stderr Type=Segmentation error vmState=0x00000000
MT5 stderr J9Generic_Signal_Number=00000004 Signal_Number=0000000b Error_Value=00000000 Signal_Code=00000032
MT5 stderr Handler1=09001000B8CEF448 Handler2=09001000B8CC6538
MT5 stderr R0=00000000E0005A38 R1=00000001100421A0 R2=00000000300215F0 R3=00000000FFF00250
MT5 stderr R4=0000000030008200 R5=000000000000000E R6=00000000FFF00250 R7=09001000B8D4D1D8
MT5 stderr R8=0000000000000000 R9=0000000000000000 R10=0000000000000002 R11=09000003545BCDA0
MT5 stderr R12=0000000042404028 R13=000000011004E800 R14=0000000030007D00 R15=0000000030008200
MT5 stderr R16=0000000120FF78F8 R17=FFFFFFFFFFFFFFFF R18=0000000110042460 R19=0A000100000F8DC4
MT5 stderr R20=000000011015E2D8 R21=0000000030008200 R22=0000000110042480 R23=00000001100574E8
MT5 stderr R24=090000035423C752 R25=0000000000000008 R26=0000000000000000 R27=09001000B8CF2DF0
MT5 stderr R28=0000000110042480 R29=09001000B8CB1698 R30=0000000030026300 R31=0000000000000000
MT5 stderr IAR=000000011052AA78 LR=09000003545BC454 MSR=A00000000000D032 CTR=000000011052AA5C
MT5 stderr CR=2240408000000004 FPSCR=8202000000000000 XER=0000000482020000
MT5 stderr FPR0 0000000000000055 (f: 85.000000, d: 4.199558e-322)
MT5 stderr FPR1 c3e0000000000000 (f: 0.000000, d: -9.223372e+18)
MT5 stderr FPR2 bfe00366d84df56c (f: 3628987648.000000, d: -5.004153e-01)
MT5 stderr FPR3 3fa47111edd8bfc1 (f: 3990405120.000000, d: 3.992516e-02)
MT5 stderr FPR4 3fa4daf09aaef7c5 (f: 2595158016.000000, d: 4.073288e-02)
MT5 stderr FPR5 3fda7aecbeba4e46 (f: 3199880704.000000, d: 4.137527e-01)
MT5 stderr FPR6 3fd5555555555611 (f: 1431655936.000000, d: 3.333333e-01)
MT5 stderr FPR7 3f5b2f0d6d4eb1a5 (f: 1833873792.000000, d: 1.659167e-03)
MT5 stderr FPR8 bfb96bfae093ca0d (f: 3767781888.000000, d: -9.930389e-02)
MT5 stderr FPR9 40fb39456ab53250 (f: 1790259840.000000, d: 1.115083e+05)
MT5 stderr FPR10 407816625220ec6e (f: 1377889408.000000, d: 3.853990e+02)
MT5 stderr FPR11 3fbc73cd95d5986d (f: 2513803264.000000, d: 1.111420e-01)
MT5 stderr FPR12 3ff0000000000000 (f: 0.000000, d: 1.000000e+00)
MT5 stderr FPR13 0000000000000003 (f: 3.000000, d: 1.482197e-323)
MT5 stderr FPR14 0000000000000000 (f: 0.000000, d: 0.000000e+00)
MT5 stderr FPR15 0000000000000000 (f: 0.000000, d: 0.000000e+00)
MT5 stderr FPR16 0000000000000000 (f: 0.000000, d: 0.000000e+00)
MT5 stderr FPR17 0000000000000000 (f: 0.000000, d: 0.000000e+00)
MT5 stderr FPR18 0000000000000000 (f: 0.000000, d: 0.000000e+00)
MT5 stderr FPR19 0000000000000000 (f: 0.000000, d: 0.000000e+00)
MT5 stderr FPR20 0000000000000000 (f: 0.000000, d: 0.000000e+00)
MT5 stderr FPR21 0000000000000000 (f: 0.000000, d: 0.000000e+00)
MT5 stderr FPR22 0000000000000000 (f: 0.000000, d: 0.000000e+00)
MT5 stderr FPR23 0000000000000000 (f: 0.000000, d: 0.000000e+00)
MT5 stderr FPR24 0000000000000000 (f: 0.000000, d: 0.000000e+00)
MT5 stderr FPR25 0000000000000000 (f: 0.000000, d: 0.000000e+00)
MT5 stderr FPR26 0000000000000000 (f: 0.000000, d: 0.000000e+00)
MT5 stderr FPR27 0000000000000000 (f: 0.000000, d: 0.000000e+00)
MT5 stderr FPR28 0000000000000000 (f: 0.000000, d: 0.000000e+00)
MT5 stderr FPR29 0000000000000000 (f: 0.000000, d: 0.000000e+00)
MT5 stderr FPR30 0000000000000000 (f: 0.000000, d: 0.000000e+00)
MT5 stderr FPR31 0000000000000000 (f: 0.000000, d: 0.000000e+00)
MT5 stderr 
MT5 stderr Compiled_method=java/lang/Object.hashCode()I
MT5 stderr Target=2_90_20190215_1097 (AIX 7.1)
MT5 stderr CPU=ppc64 (8 logical CPUs) (0x200000000 RAM)
MT5 stderr ----------- Stack Backtrace -----------
MT5 stderr sendClinit+0x1f0 (0x09000003540B2D94 [libj9vm29.so+0x53d94])
MT5 stderr initializeImpl+0x3fc (0x09000003540CD980 [libj9vm29.so+0x6e980])
MT5 stderr classInitStateMachine__FP10J9VMThreadP7J9Class16J9ClassInitState+0xda8 (0x09000003540CEACC [libj9vm29.so+0x6facc])
MT5 stderr resolveClassRef+0x3d4 (0x09000003540AB6B8 [libj9vm29.so+0x4c6b8])
MT5 stderr (0x0900000354234C5C [libj9vm29.so+0x1d5c5c])
MT5 stderr (0x0900000354117CFC [libj9vm29.so+0xb8cfc])
MT5 stderr sendClinit+0x1f0 (0x09000003540B2D94 [libj9vm29.so+0x53d94])
MT5 stderr initializeImpl+0x3fc (0x09000003540CD980 [libj9vm29.so+0x6e980])
MT5 stderr classInitStateMachine__FP10J9VMThreadP7J9Class16J9ClassInitState+0xda8 (0x09000003540CEACC [libj9vm29.so+0x6facc])
MT5 stderr resolveClassRef+0x3d4 (0x09000003540AB6B8 [libj9vm29.so+0x4c6b8])
MT5 stderr (0x0900000354234C5C [libj9vm29.so+0x1d5c5c])
MT5 stderr (0x0900000354117CFC [libj9vm29.so+0xb8cfc])
MT5 stderr sendClinit+0x1f0 (0x09000003540B2D94 [libj9vm29.so+0x53d94])
MT5 stderr initializeImpl+0x3fc (0x09000003540CD980 [libj9vm29.so+0x6e980])
MT5 stderr classInitStateMachine__FP10J9VMThreadP7J9Class16J9ClassInitState+0xda8 (0x09000003540CEACC [libj9vm29.so+0x6facc])
MT5 stderr resolveStaticMethodRefInto+0x1fc (0x09000003540ABF60 [libj9vm29.so+0x4cf60])
MT5 stderr resolveStaticMethodRef+0x30 (0x09000003540AC494 [libj9vm29.so+0x4d494])
MT5 stderr (0x0900000354218284 [libj9vm29.so+0x1b9284])
MT5 stderr (0x0900000354117CFC [libj9vm29.so+0xb8cfc])
MT5 stderr initializeAttachedThreadImpl+0x39c (0x09000003540B29C0 [libj9vm29.so+0x539c0])
MT5 stderr initializeAttachedThread+0x5c (0x09000003540B92A0 [libj9vm29.so+0x5a2a0])
MT5 stderr standardInit+0x288 (0x0900000355E6366C [libjclse29.so+0x3d66c])
MT5 stderr J9VMDllMain+0x314 (0x0900000355EAED58 [libjclse29.so+0x88d58])
MT5 stderr runJ9VMDllMain@AF73_47+0xf4 (0x0900000354072BD8 [libj9vm29.so+0x13bd8])
MT5 stderr pool_do+0xf8 (0x090000035407DDDC [libj9vm29.so+0x1eddc])
MT5 stderr IPRA.$runInitializationStage+0xc8 (0x0900000354070BEC [libj9vm29.so+0x11bec])
MT5 stderr protectedInitializeJavaVM+0xd70 (0x0900000354069E74 [libj9vm29.so+0xae74])
MT5 stderr omrsig_protect+0x3cc (0x0900000354367670 [libj9prt29.so+0x69670])
MT5 stderr initializeJavaVM+0x218 (0x0900000354068BFC [libj9vm29.so+0x9bfc])
MT5 stderr J9_CreateJavaVM+0xa4 (0x0900000354087628 [libj9vm29.so+0x28628])
MT5 stderr JNI_CreateJavaVM+0xa0c (0x090000035400DF70 [libjvm.so+0x28f70])
MT5 stderr JNI_CreateJavaVM+0x1c4 (0x0900000353FCFA48 [libjvm.so+0x24a48])
MT5 stderr (0x0000000100003940 [java+0x3940])
MT5 stderr _pthread_body+0xf0 (0x0900000000526E14 [libpthreads.a+0x3e14])
```

**Comment by R-Santhir** [comment]
Having talked to Devin, I think I've located the spot in the JCL source where the failure occurs:

```
0x110529cdc  +120  -1:65        e8500f18 ld       r2,+3864(r16) JITTOC 0x300213f0 = 0x4  <<< +49
0x110529ce0  +121               80620000 lwz      r3,+0(r2) <<< ^+235
0x110529ce4  +122               60000000 mr       r0,r0
0x110529ce8  +123               2c030004 cmpi     cr0,0,r3,+4 <<< a comparison
0x110529cec  +124          |    4082009c bne-     cr0,0x110529d88 C>> +163
0x110529cf0  +125          |    e86e0050 ld       r3,+80(r14)  <<<<<<<<<<<LOADING BOTH r3 and r0
0x110529cf4  +126          |    e80e0060 ld       r0,+96(r14)
0x110529cf8  +127          |    7c63002e lwzx     r3,r3,r0 <<<<<<<<<< Failure point
```

`
long ptr = (com.ibm.oti.vm.VM.FJ9OBJECT_SIZE == 4) ? Integer.toUnsignedLong(h.getIntFromObject(anObject, 0L)) : h.getLongFromObject(anObject, 0L);
`

It looks like the above source corresponds to the comparison that occurs in the assembly. So the problem looks like it lies in the call to `Integer.toUnsignedLong(h.getIntFromObject(anObject, 0L))`   I will investigate it further.

**Comment by gita-omr** [comment]
A brief update: @R-Santhir was able to generate a tracefile for this method, but at that time it did not fail and the tracefile looks very different from the core. In particular, the load from the object looks normal (no adding helper address to it).  @liqunl maybe you know what could cause getIntFromObject to add helper address to the object? 

@fjeremic @dchopra001

**Comment by liqunl** [comment]
Looking at the disassembly of the failing method, we're trying to load from `(anObject + jitHelpers)` and pass it to `Integer.toUnsignedLong`. The load maps to `h.getIntFromObject(anObject, 0L)` which should be transformed to a iloadi from `anObject`. The other `h.getIntFromObject(anObject, 0L)` was transformed to the correct load. `+96(r14)` is the `anObject` on the stack. 

```
+163          >||| e86e0060 ld       r3,+96(r14) <<< +124
+164           ||| e8630000 ld       r3,+0(r3)
```

In my unit test on aix with java8, trees generated by UnsafeFastPath look good. It could be other opts or the codegen that made the load wrong. We need a trace log to continue the investigation.

**Comment by gita-omr** [comment]
So we are back to square one. Even though we know what is wrong with the generated code nobody knows why it was generated. So we need to reproduce the original failure with original options and then with a tracefile.

**Comment by gita-omr** [comment]
@R-Santhir was able to get a tracefile for a similar defect and it definitely points to the Unsafe Fast Path optimization so removing comp:jit:p. @andrewcraik @fjeremic

**Comment by andrewcraik** [comment]
yup @liqunl is on it

**Comment by liqunl** [comment]
@jdmpapin is working on the fix.

**Comment by pshipton** [comment]
@jdmpapin @andrewcraik this is tagged against the 0.13 milestone. Is there a PR to fix the problem in the 0.13 branch? If not please cherry pick the change and create one.

**Comment by andrewcraik** [comment]
@jdmpapin can you please create a PR against the 0.13 milestone?
